modules/microshift-data-cleaning-container-images.adoc

```diff
@@ -0,0 +1,86 @@
+// Module included in the following assemblies:
+//
+// * microshift_troubleshooting/microshift-cleanup-data.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-data-cleaning-container-images_{context}"]
+= Cleaning all data and keeping the container images
+
+Retain the {microshift-short} container images while cleaning all data by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` arguments.
+
+Keeping the container images helps speed up {microshift-short} restart after data cleanup because the necessary container images are already present locally when you start the service.
+
+When you run the script with the `--all` and `--keep-images` arguments, you perform the following cleanup actions:
+
+* All {microshift-short} services are stopped and disabled
+* Deletes all {microshift-short} pods
+* Resets network configuration
+* Deletes the `/var/lib/microshift` data directory
+* Deletes OVN-K networking configuration
+
+[WARNING]
+====
+This option deletes all {microshift-short} data and user workloads. Use with caution.
+====
+
+.Prerequisites
+. You are logged into the {microshift-short} as an administrator with root-user access.
+. You have filed a support case.
+
+.Procedure
+
+. Cleanup all data and user workloads while retaining the {microshift-short} container images by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` argument, by entering the following command:
++
+[source,terminal]
+----
+$ sudo microshift-cleanup-data --all --keep-images
+----
++
+.Example output
+[source,terminal]
+----
+DATA LOSS WARNING: Do you wish to stop and clean ALL MicroShift data AND cri-o container workloads?
+1) Yes
+2) No
+#? Yes
+Stopping MicroShift services
+Disabling MicroShift services
+Removing MicroShift pods
+Deleting the br-int interface
+Killing conmon, pause and OVN processes
+Removing MicroShift configuration
+Removing OVN configuration
+MicroShift service was stopped
+MicroShift service was disabled
+Cleanup succeeded
+
+$ sudo crictl images | awk '{print $1}'
```

Comment:
Just checking - is this legitimately part of the script output or is this an a command that users need to run?

Reply:
We could add something like:
Verify that the container images are still present after running the `microshift-cleanup-data` script.

```
$ sudo crictl images | awk '{print $1}'
IMAGE
quay.io/openshift-release-dev/ocp-v4.0-art-dev
quay.io/openshift-release-dev/ocp-v4.0-art-dev
...
...
```